### DISCUSSION TITLE
How do I start an Astro + Pages Function dev server if `-- <command>` is being deprecated?

### DISCUSSION THREAD

**Discussion by AVGVSTVS96** [cloudflare/workers-sdk#5617] [General]
[Cloudflare Wrangler Deprecations](https://developers.cloudflare.com/workers/wrangler/deprecations/) states that `-- <command>, --proxy, and --script-path` in wrangler pages dev will be deprecated soon.

### My Application
I have an Astro app with a GPT chat page which uses a pages function to communicate with OpenAI and my frontend. I use Bun as a package manager and am on Wrangler 3.48.0.

### My current workflow

I currently start my dev environment by running `bunx wrangler pages dev -- bun run dev` to start a dev server for both my pages function and Astro frontend. I then navigate to localhost:8787 and am able to access both my UI and, through it, the OpenAI API using the CF pages function. This allows me to test my full application as if it was deployed on Cloudflare.

#### If this is deprecated, how will I be able to run a dev server? 

[Cloudflare Wrangler Deprecations](https://developers.cloudflare.com/workers/wrangler/deprecations/) states:

> Instead of relying on Wrangler to proxy through to some other upstream dev server, you can emulate a more accurate behavior by building your static assets to a directory and pointing Wrangler to that directory with wrangler pages dev <directory>

So I will have to rebuild my Astro app and run the command in `dist` directory every time I make a change? No more hot reloading?

I tested `bunx wrangler pages dev dist`, this worked, but I have to rebuild and restart the server after every change.

`-- bun run dev` is the only way I know of to run my frontend dev server and allow it to communicate with my backend through Wrangler, as if it's deployed on CF while enabled hot reloading.

I may be missing something because I'm a new programmer and am still learning about CF and Wrangler, I haven't seen much documentation applicable to this scenario. Any help would be greatly appreciated!

I'm hoping theres still a way to test my app in a standard dev environment after `-- <command>` is deprecated, I really like how minimal it is to use a pages function to handle OpenAI API communication in my Astro app and this is the only way I know of to test it without having to rebuild and restart after every change.

**Comment by malobre** [comment]
You need to run your build command with some kind of watch mode (i.e, rebuild on file change), and run `wrangler pages dev dist` in parallel (where `dist` is the output directory of your build command).

This was mentioned in the deprecation issue: https://github.com/cloudflare/workers-sdk/issues/5211#issuecomment-2008192443

I'm not familiar with `bun` but looking at [the documentation](https://bun.sh/docs/cli/run#watch), if your build script is named "build" `bun --watch run build` should work ?

Edit: Note that live-reload is currently broken (https://github.com/cloudflare/workers-sdk/issues/5351)

**Reply by AVGVSTVS96** [reply]
Thank you for your response. Does live-reload not working mean `--watch` won't work?

I don't understand why this is being deprecated, this seems to be creating a lot more trouble than it's worth and will degrade Cloudflare's DX significantly. Is there any chance that CF will reverse this?

**Reply by malobre** [reply]
It mean that `wrangler pages dev --live-reload dist` doesn't auto reload your browser tab on file change.

Regarding the reversal of the deprecation I don't know, I don't work for cloudflare I'm just another user.
